tests/test_client.py

```diff
@@ -0,0 +1,382 @@
+
+    options: EnqueueOptions = {
+        "queue_name": "test_queue",
+        "workflow_name": "enqueue_test",
+        "workflow_id": wfid,
+    }
+
+    client.enqueue(options, 42, "test", johnDoe)
+
+    handle: WorkflowHandle[str] = DBOS.retrieve_workflow(wfid)
+    result = handle.get_result()
+    assert result == '42-test-{"first": "John", "last": "Doe", "age": 30}'
+
+    wf_status = dbos.get_workflow_status(wfid)
+    assert wf_status is not None
+    assert wf_status.status == "SUCCESS"
+    assert wf_status.name == "enqueue_test"
+    assert wf_status.app_version == GlobalParams.app_version
+
+
+def test_client_enqueue_appver_set(dbos: DBOS, client: DBOSClient) -> None:
+    run_client_collateral()
+
+    johnDoe: Person = {"first": "John", "last": "Doe", "age": 30}
+    wfid = str(uuid.uuid4())
+
+    options: EnqueueOptions = {
+        "queue_name": "test_queue",
+        "workflow_name": "enqueue_test",
+        "workflow_id": wfid,
+        "app_version": GlobalParams.app_version,
+    }
+
+    client.enqueue(options, 42, "test", johnDoe)
+
+    handle: WorkflowHandle[str] = DBOS.retrieve_workflow(wfid)
+    result = handle.get_result()
+    assert result == '42-test-{"first": "John", "last": "Doe", "age": 30}'
+
+    wf_status = dbos.get_workflow_status(wfid)
+    assert wf_status is not None
+    assert wf_status.status == "SUCCESS"
+    assert wf_status.name == "enqueue_test"
+    assert wf_status.app_version == GlobalParams.app_version
+
+
+def test_client_enqueue_wrong_appver(dbos: DBOS, client: DBOSClient) -> None:
+    run_client_collateral()
+
+    johnDoe: Person = {"first": "John", "last": "Doe", "age": 30}
+    wfid = str(uuid.uuid4())
+
+    options: EnqueueOptions = {
+        "queue_name": "test_queue",
+        "workflow_name": "enqueue_test",
+        "workflow_id": wfid,
+        "app_version": "0123456789abcdef",
+    }
+
+    client.enqueue(options, 42, "test", johnDoe)
+
+    time.sleep(5)
+
+    wf_status = dbos.get_workflow_status(wfid)
+    assert wf_status is not None
+    assert wf_status.status == "ENQUEUED"
+    assert wf_status.name == "enqueue_test"
+    assert wf_status.app_version == options["app_version"]
+
+
+def test_client_enqueue_idempotent(
+    config: ConfigFile, client: DBOSClient, sys_db: SystemDatabase
+) -> None:
+    DBOS.destroy(destroy_registry=True)
+
+    johnDoe: Person = {"first": "John", "last": "Doe", "age": 30}
+    wfid = str(uuid.uuid4())
+
+    options: EnqueueOptions = {
+        "queue_name": "test_queue",
+        "workflow_name": "enqueue_test",
+        "workflow_id": wfid,
+    }
+
+    client.enqueue(options, 42, "test", johnDoe)
+    client.enqueue(options, 42, "test", johnDoe)
+
+    wf_status = sys_db.get_workflow_status(wfid)
+    assert wf_status is not None
+    assert wf_status["status"] == "ENQUEUED"
+    assert wf_status["name"] == "enqueue_test"
+    assert wf_status["app_version"] == None
+
+    dbos = DBOS(config=config)
+    DBOS.launch()
```

Comment:
Just curious: why do you need to launch DBOS later for this test?

Reply:
So I could enqueue twice and be able to ensure only a single message was enqueued before it was picked up for processing